interview-questions.md

```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+Your answer:
+
+- The foreign key will be added in Student model by adding each students with unique key.
+
+Researched answer:
+
+- I made a visual diagram to help visualize the prompt and figue out where the foreign key would go. Through this i came to my senses that the propmt said that the Cohort model has_many students which means we need to create an association migration. It also made sense that the Cohort model is the primry key and the Student Model is where the the list of the students will go and can update student_id foreign key that's assigned as unique values to each students not add the student keys to the Cohort model because it's the primary key and theres only one Cohort.
 
-  Researched answer:
+Visual diagram:
 
+Cohort Model:
+Title: String
+id: integer
 
+cohort1 = new Cohort()
+Title: Alpha
+Id: 1 (primarykey)
+
+Cohort -> (has_many) Students
+
+Student Model:
+Name: string
+Age: integer
+cohort_id: student_id (foreignkey)
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- Thinking from what i remember about the RESTful routes i think it will be the update method because that's when it will passed params to make changes.
 
+Researched answer:
 
+- Through my research I learned that the RESTful routes that must always passed params are request GET, PATCH and DELETE because these actions requires passing params throughout the database after doing the CRUD actions.
```

Comment:
Please review the section in the syllabus that gives more detail about which restful routes and controller action need params. 
https://github.com/learn-academy-2022-alpha/Syllabus/blob/main/rails/restful-routes-crud.md